pkg/scalers/gcp_storage_scaler.go

```diff
@@ -204,22 +204,22 @@ func (s *gcsScaler) getItemCount(ctx context.Context, maxCount int64) (int64, er
 
 	for count < maxCount {
 		item, err := it.Next()
-		if err == iterator.Done {
-			break
-		}
-		// The folder is retrieved as an entity, so if size is 0
-		// we can skip it
-		if item.Size == 0 {
-			continue
-		}
 		if err != nil {
+			if errors.Is(err, iterator.Done) {
```

Comment:
Why do we have this unrelated change here?

Reply:
moving `item.Size == 0` to after checking the error is to prevent another panic (only `err == iterator.Done` has checked and not any error), the change of `err == iterator.Done` to inside the if statement is to evaluate all the errors together with the same format, but I can revert that part and do it as part of another PR to be clearer if you want

Reply:
np